proxmox.go

```diff
@@ -128,7 +129,7 @@ func (c *Client) Req(ctx context.Context, method, path string, data []byte, v in
 			return ErrNotAuthorized
 		}
 
-		if c.credentials != nil && c.session == nil {
+		if c.credentials != nil && (c.autoReLogin || c.session == nil) {
```

Comment:
maybe im over thinking this, but as written i believe if you don't have access to an endpoint this will continue to try and relogin and attempt the request over and over again putting you into an infinite loop.

you might want to try this against a limited user with no access to something and make a request to it... if it does then we probably want to have some form of a max relogin attempts and return an ErrNotAuthorized when you reach it

Reply:
```go
if res.StatusCode == http.StatusUnauthorized || res.StatusCode == http.StatusForbidden {
	if path == (c.baseURL + "/access/ticket") {
		// received an unauthorized while trying to create a session
		return ErrNotAuthorized
	}

	if c.credentials != nil && (c.autoReLogin || c.session == nil) {
		// credentials passed but no session started, try a login and retry the request
		if _, err = c.Ticket(ctx, c.credentials); err != nil {
			return err
		}
		return c.Req(ctx, method, path, data, v)
	}
	return ErrNotAuthorized
}
```
Sorry, but I don't think this change will loop indefinitely.

If `autoReLogin` is triggered, it will try to re-acquire the Ticket, but if URL is `/access/ticket` and an unauthorized status code is returned, will immediately return `ErrNotAuthorized`.

I added this option to maintain the instance session, which means that the instance must have permission to access the endpoint when it is initialized, and the session will be automatically renewed later.

It is possible to add a maximum number of attempts. I am considering how to implement this feature elegantly.

Reply:
hard for me to explain, sorry... i think if you login the client and then make a request to something which returns unauthorized or forbidden the auto relogin will force the relogin which will succeed and then retry the request forcing the infinite loop. i think the safest thing is to do a retry limit and return a ErrNotAuthorized if relogin fails x amount of times. happy to hear if you disagree